src/WaterCost.jl

```diff
@@ -0,0 +1,110 @@
+using Mimi
+using Distributions
+
+include("lib/datastore.jl")
+include("watercostdata.jl")
+
+@defcomp WaterCost begin
+    regions = Index()
+    canals = Index()
+    aquifers = Index()
+
+    # UNIT COST
+    # extraction COST
+    unitgwextractioncost = Parameter(index=[aquifers, time], unit="\$/1000 m^3")
+    unitswextractioncost = Parameter(index=[canals, time], unit="\$/1000 m^3")
+    # treatment cost
+    unitgwtreatmentcost = Parameter(index=[regions, time], unit="\$/1000 m^3")
+    unitswtreatmentcost = Parameter(index=[regions, time], unit="\$/1000 m^3")
+    # distribution cost
+    unitdistributioncost = Parameter(index=[regions, time], unit="\$/1000 m^3")
+    
+    # Supersource
+    unitsupersourcecost = Parameter(index=[regions, time], unit="\$/1000 m^3")
+
+    # VOLUMES 
+    swwithdrawals = Parameter(index=[canals, time], unit="1000 m^3")
+    gwextraction = Parameter(index=[aquifers, time], unit="1000 m^3")
+    supersourcesupply = Parameter(index=[regions, time], unit="1000 m^3")
+    # COST
+    gwcost = Variable(index=[regions, time], unit="\$/1000 m^3")
+    swcost = Variable(index=[regions, time], unit="\$/1000 m^3")
+    supersourcecost = Variable(index=[regions, time], unit="\$/1000 m^3")
+    totalcost = Variable(index=[regions, time], unit="\$/1000 m^3")
+    
+end
+
+"""
+Sets the cost of the water in function of the state of the system
+"""
+function run_timestep(c::WaterCost, tt::Int)
+    v = c.Variables
+    p = c.Parameters
+    d = c.Dimensions
+
+    v.swcost[:,tt] = zeros(numcounties)
+    for pp in 1:nrow(draws)
+	    fips = draws[pp, :fips] < 10000 ? "0$(draws[pp, :fips])" : "$(draws[pp, :fips])"
+	    rr = findfirst(mastercounties[:fips] .== fips)
+	    if rr > 0
+		    v.swcost[rr,tt] += p.swwithdrawals[pp,tt]*(p.unitswextractioncost[pp,tt]+p.unitswtreatmentcost[rr,tt]+p.unitdistributioncost[rr,tt])
+	    end
```

Comment:
We should add a column to draws that already has the correct rr values, so that we don't have to do any searches and string operations here, which are both comparatively slow.